src/services/sync/strategies/consumerActions/BaseConsumerAction.js

```diff
@@ -0,0 +1,86 @@
+const { NotImplemented, SyncError, BadRequest } = require('../../../../errors');
+const { batchFilterKeys } = require('../../utils');
+
+const securedKeys = ['type', 'getType', 'matchType', 'filterData', 'exec', 'action', 'getSecuredKeys'];
+
+class BaseConsumerAction {
+	constructor(type, options) {
+		if (!type) {
+			throw new NotImplemented('The type must set for consumer actions.');
+		}
+		this.filterActive = true;
+		this.type = type;
+		this.allowedLogKeys = null;
+
+		Object.entries(options).forEach(([k, v]) => {
+			if (!securedKeys.includes(k)) {
+				this[k] = v;
+			}
+		});
```

Comment:
Why do you need any possible attribute to be set here except of securedKeys? Currently we have only `allowedLodgKeys` . Whe could set this attribute directly if it is available.